src/app/entity/Bet.ts

```diff
@@ -1,9 +1,11 @@
 import { UserProfile } from './UserProfile';
+import { CompleteBetType } from './complete-bet-type.model';
 
 export interface Bet {
   betId?: number;
-  user?: UserProfile;
+  user?: UserProfile | null;
   amount: number;
-  odds: number;
+  odds?: number;
```

Comment:
Why do you store the odds in the Bet ? Shouldn't those be stored in the CompleteBetType ?